docs/tumblerguide.md

```diff
@@ -1,12 +1,122 @@
-## Running the tumbler
+## Running the tumbler.
 
-Assuming installation has been done correctly, running the `tumbler.py` script in the `scripts/` directory is almost the same as that in the [original codebase](https://github.com/Joinmarket-Org/joinmarket).
+The information in this guide is supplemental to that in the usage guide. Note that the tumbler can be run as a script:
 
-As a reminder, the explanation of the principal details of how to do this are in [the wiki](https://github.com/JoinMarket-Org/joinmarket/wiki/Step-by-step-running-the-tumbler).
+```
+(jmvenv)a@~/joinmarket-clientserver/scripts$ python tumbler.py --help
+```
+
+or from the JoinmarketQt app in the "Multiple Joins" tab (see [the guide](JOINMARKET-QT-GUIDE.md)).
+
+# Contents
+
+1. [Introduction to the tumbler](#introduction)
+
+   a. [A note on fees](#a-note-on-fees)
+
+2. [Basic examples](#basic)
+
+   a. [Example: Tumbling into your wallet after buying from an exchange to improve privacy](#example-tumbling-into-your-wallet-after-buying-from-an-exchange-to-improve-privacy)
+
+   b. [Example: Tumbling from your wallet into an exchange](#example-tumbling-from-your-wallet-into-an-exchange)
+
+3. [Schedules (transaction lists)](#schedules)
+
+4. [Tumbler schedule and logging](#logging)
+
+5. [Interpreting key console output](#interpreting)
+
+6. [Tweaking the schedule](#tweaking)
+
+7. [How often do retries occur?](#how-often-do-retries-occur)
+
+8. [Restarts](#restarts)
+
+9. [Possible failure vectors](#failure-vectors)
+
+<a name="introduction" />
+
+## Introduction to the tumbler
+
+Tumbler is a JoinMarket bot which attempts to completely break the link between addresses. It is used to restore privacy where it has been damaged. It creates many many coinjoins to bounce coins around in different amounts and times.
+
+Examples of users might be people who bought bitcoins with a very privacy-invading method, such as buying from an exchange, and wish to have privacy in all their purchases again. Some bitcoin users also just need it as a simple medium of exchange, buying bitcoins with traceable fiat and immediately spending them on goods and services. Example would be an anonymous buyer of a domain name, VPS hosting, email or VPN provisions. Users also might be those who engage in capital flight or want to store bitcoins without anyone knowing, tumbling them into cold storage. If bitcoin fungibility is ever attacked the messages "Your coins are rejected because they were used for illegal or immoral activity X transactions ago" then this bot can probably be used to solve the problem.
```

Comment:
This is one for @chris-belcher ... I did consider maybe either removing or more likely editing this sentence, but it's carried over from the wiki, which was previously linked here. Personally I don't have a problem parsing the sentence, per se, just not sure it's appropriate.

Reply:
What is unclear to me is how we go from `If bitcoin fungibility is ever attacked the messages` to `then this bot`.

But it could easily be a bad understanding by me.

Reply:
Oh, apologies, I somehow had a blind spot and didn't notice that the sentence doesn't scan. Curiously it is written like that in the original: https://github.com/JoinMarket-Org/joinmarket/wiki/Step-by-step-running-the-tumbler . I'll let @chris-belcher answer as to what he wants to do here.

Reply:
How about something like: `If bitcoin fungibility is ever attacked, leading to users being faced with messages like "Your coins are rejected because they were used for illegal or immoral activity X transactions ago" then the tumbler app can probably solve the problem.`

Reply:
Will do.

Reply:
Done in https://github.com/JoinMarket-Org/joinmarket-clientserver/pull/644/commits/35460ffd28e6c7392417784de7fab2d7c7bf4a66